core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -350,20 +350,16 @@ private[spark] class TaskSchedulerImpl(
                 removeExecutor(execId, reason.get)
                 failedExecutor = Some(execId)
               }
+            }
+            if (TaskState.isFinished(state)) {
+              cleanupTaskState(tid)
```

Comment:
I don't think that's necessary because all access to the TSM is gated on the TaskSchedulerImpl, so even though the TaskResultGetter might do some stuff, the TSM's state won't be accessed until the later handleSuccessfulTask call to the TaskSchedulerImpl